app/grandchallenge/profiles/models.py

```diff
@@ -55,9 +67,13 @@ class UserProfile(models.Model):
         help_text="Display the organizations that you are a member of in your profile.",
     )
 
-    receive_notification_emails = models.BooleanField(
-        default=True,
-        help_text="Whether to receive notification emails",
+    receive_notification_emails = models.CharField(
```

Comment:
Unfortunately we cannot just alter the field. There will be code running after the migration has happened but is still assuming that a boolean field is in use. This will only be 10 minutes but our application generates a lot of notifications. With a char field all of the `if self.receive_notification_emails` will pass, and we would end up emailing people who have opted out.

The approach needs to change:

- a new field should be added to store the choice (maybe `notification_email_choice`?), this will be one migration file.
- a data migration (e.g. a second migration file as AFAIK db modification and data changes cannot happen in the same transaction in postgres) should be made to fill out the new field
- the code should be updated to use the new field (search the codebase to ensure there are no more instances of `receive_notification_emails`)

That's 1 PR.

Once that is deployed a 2nd PR needs to be created where `receive_notification_emails` should be marked as deprecated with `deprecate_field` and a new migration made.

Then, once that is deployed a 3rd PR should be created to remove the `receive_notification_emails` field and a new migration made.

I won't check the rest of the PR right now as unfortunately this is a lot to change, sorry.

Reply:
Ok, I understand the step wise approach, and understand that this is generally how one should ago about making these types of db schema changes, but in this specific case, I (maybe wrongly) reasoned that the 10 minutes or so after the migration runs and before the new code is running don't matter since we only send the email once a day at 4am and so by the time we send emails, the new code would be running (instant email wouldn't have an effect since no user has that set as their preference after migrating). Does that make sense?

I will change the approach, just wanted to explain why I did what I did. :-)